### PR TITLE
Keeps live stream open for empty persistence ids #28428

### PR DESCRIPTION
**Pull request by nvollmar**
Refs #28428

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

**Comment by nvollmar** [comment]
@patriknw May I ask you to have a look? It would be great if we could get that fixed in Akka 2.6.3.
Since upgrading to Akka 2.6 we have stability issues with our tests which I suspect akka-persistence-inmemory to be the cause of. Switching to LevelDB (and Akka patched with this PR) I cannot reproduce those issues anymore.

**Review by johanandren — APPROVED** [review]
LGTM, thanks @nvollmar

File: akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdSpec.scala [hunk 1]
```diff
@@ -92,7 +91,7 @@ class EventsByPersistenceIdSpec extends AkkaSpec(EventsByPersistenceIdSpec.confi
       expectMsg(s"${3L}-deleted")
 
       val src = queries.currentEventsByPersistenceId("g1", 0L, Long.MaxValue)
-      src.map(_.event).runWith(TestSink.probe[Any]).request(1).expectComplete()
+      src.map(_.event).runWith(TestSink.probe[Any]).request(1).cancel()
```
Comment (raboof) [inline akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdSpec.scala]: Does this still test anything now?

**Review by raboof — APPROVED** [review]
Seems reasonable to me

File: akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdSpec.scala (hunk 1 above)
Comment (nvollmar) [reply akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdSpec.scala]: I will have a look at it

Comment (nvollmar) [reply akka-persistence-query/src/test/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdSpec.scala]: It was an error. I've reverted it and pushed an update.

**Comment by nvollmar** [comment]
It should be now ready to merge - would be great to have it in the next Akka release 👍

**Comment by nvollmar** [comment]
Rebased to latest master to resolve conflict

**Pull request merged**